server/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java

```diff
@@ -176,4 +178,43 @@ void givenTwoTaskManagerATaskSubmittedOnOneCouldBeRunOnTheOther() throws Interru
         assertThat(executionDetails.getSubmittedNode()).isEqualTo(HOSTNAME_2);
         assertThat(executionDetails.getRanNode()).contains(HOSTNAME);
     }
+
+    private List<TaskId> extractIds(List<TaskExecutionDetails> tasks) {
+        return tasks.stream()
+            .map(TaskExecutionDetails::getTaskId)
+            .collect(Guavate.toImmutableList());
+    }
+
+    private boolean allTasksAreCompleted(List<TaskExecutionDetails> tasks) {
+        List<TaskExecutionDetails> completedTasks = tasks.stream()
+            .filter(task -> task.getStatus().equals(TaskManager.Status.COMPLETED))
+            .collect(Guavate.toImmutableList());
+
+        return completedTasks.size() == tasks.size();
+    }
+
+    @Test
+    void givenTwoTaskManagerAndATaskRanPerTaskManagerListingThemOnEachShouldShowBothTasks() throws InterruptedException {
+        TaskManager taskManager1 = taskManager(HOSTNAME);
+        TaskManager taskManager2 = taskManager(HOSTNAME_2);
+
+        TaskId taskId1 = taskManager1.submit(new CompletedTask());
+        TaskId taskId2 = taskManager2.submit(new CompletedTask());
+
+        ImmutableSet<TaskId> expectedTaskIds = ImmutableSet.of(taskId1, taskId2);
+        int expectedNumberOfTasks = 2;
+
+        Awaitility.await()
+            .atMost(Duration.ONE_SECOND)
+            .pollInterval(100L, TimeUnit.MILLISECONDS)
+            .until(() -> {
```

Comment:
I think you can achieve a more readable output by relying on "untilAsserted" + dump regular assertj statements...